iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -265,6 +273,36 @@ public String getFileFormatPropertyKey() {
     return TableProperties.DEFAULT_FILE_FORMAT;
   }
 
+  @Override
+  public boolean commitInMoveTask() {
+    return true;
+  }
+
+  @Override
+  public void storageHandlerCommit(Properties commitProperties, boolean overwrite) throws HiveException {
+    String tableName = commitProperties.getProperty(Catalogs.NAME);
+    Configuration configuration = SessionState.getSessionConf();
+    Optional<JobContext> jobContext = generateJobContext(configuration, tableName, overwrite);
+    if (jobContext.isPresent()) {
+      OutputCommitter committer = new HiveIcebergOutputCommitter();
+      try {
+        // Committing the job
+        committer.commitJob(jobContext.get());
+      } catch (Throwable e) {
+        // Aborting the job if the commit has failed
+        LOG.error("Error while trying to commit job: {}, starting rollback changes for table: {}",
+            jobContext.get().getJobID(), tableName, e);
+        try {
+          committer.abortJob(jobContext.get(), JobStatus.State.FAILED);
+        } catch (IOException ioe) {
+          LOG.error("Error while trying to abort failed job. There might be uncleaned data files.", ioe);
+          // no throwing here because the original exception should be propagated
+        }
+        throw new HiveException("Error committing job: " + jobContext.get().getJobID() + " for table: " + tableName);
```

Comment:
Can we include `e` in the exception to get the underlying cause?

Reply:
ehh... always when I try to do things fast, I make stupid mistakes....
Thanks for catching